### ISSUE TITLE
a.dropdown-toggle should be button.dropdown-toggle

### ISSUE THREAD

**Issue by fulldecent** [twbs/bootstrap#40995]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

[Current documentation](https://github.com/twbs/bootstrap/blob/c2a7d686de56f5cdb829c63e1b3849a2c04409d0/site/content/docs/5.3/examples/navbars/index.html#L28) recommends to use a dropdown toggle as:

```html
<a class="nav-link dropdown-toggle" href="#" data-bs-toggle="dropdown" aria-expanded="false">Dropdown</a>
```

But this requires the unsemantic `href="#"`. In production, I consider that to be a broken link.

Instead a better semantic could be:

```html
<button class="nav-link dropdown-toggle" data-bs-toggle="dropdown" aria-expanded="false">Dropdown</button>
```

---

This means that we would only ever include `href="#"` to recommend that a designer will link to one specific page on their website. We will not use it to mean a literal `href="#"` that is published to the web.

### Reduced test cases

```html
<nav class="navbar navbar-expand-lg bg-body-tertiary">
  <div class="container-fluid">
    <a class="navbar-brand" href="#">Navbar</a>
    <a class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </a>
    <div class="collapse navbar-collapse" id="navbarSupportedContent">
      <ul class="navbar-nav me-auto mb-2 mb-lg-0">
        <li class="nav-item">
          <a class="nav-link active" aria-current="page" href="#">Home</a>
        </li>
        <li class="nav-item">
          <a class="nav-link" href="#">Link</a>
        </li>
        <li class="nav-item dropdown">
          <button class="nav-link dropdown-toggle" href="#" role="button" data-bs-toggle="dropdown" aria-expanded="false">
            Dropdown
          </button>
          <ul class="dropdown-menu">
            <li><a class="dropdown-item" href="#">Action</a></li>
            <li><a class="dropdown-item" href="#">Another action</a></li>
            <li><hr class="dropdown-divider"></li>
            <li><a class="dropdown-item" href="#">Something else here</a></li>
          </ul>
        </li>
        <li class="nav-item">
          <a class="nav-link disabled" aria-disabled="true">Disabled</a>
        </li>
      </ul>
      <form class="d-flex" role="search">
        <input class="form-control me-2" type="search" placeholder="Search" aria-label="Search">
        <button class="btn btn-outline-success" type="submit">Search</button>
      </form>
    </div>
  </div>
</nav>
```

### What operating system(s) are you seeing the problem on?

macOS

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

5.3.3

**Comment by TheDerbiedOne** [comment]
I can take this.

I’ll open a PR updating the navbar dropdown example to use a `<button class="nav-link dropdown-toggle">` instead of an `<a href="#">`, to avoid recommending an unsemantic `href="#"` in docs/examples.

(Keeping the existing Bootstrap markup/attributes otherwise unchanged.)

**Comment by julien-deramond** [comment]
I agree that using `<button>` is generally the recommended approach. In our documentation, there are several places where `<a>` is used instead, possibly because the intended rendering was closer to a link than a button (though I'm not entirely sure of the original rationale).

Ideally, these examples should use `<button>`. However, updating some of them while preserving the current appearance may be non-trivial, which could make the resulting PR fairly substantial.

In the meantime, I've opened https://github.com/twbs/bootstrap/pull/42482 to add the missing `role="button"` attributes where applicable. I also aligned the attribute ordering across these examples to make them easier to locate and review.

**Comment by Huaian666** [comment]
the middleware approach here is way cleaner than what i was using

**Comment by mdo** [comment]
Fixed by #42552
